app/ui/browser/actions/main-actions.js

```diff
@@ -34,7 +34,16 @@ export function createTab(location: ?string = undefined,
     const id = uuid.v4();
 
     // Start loading a tab while asking the User Agent to start its browsing session.
-    dispatch({ type: types.CREATE_TAB, id, ancestorId, location, options, instrument: true });
+    // Dispatching a list ensures only one render and prevents racing to set focus across user
```

Comment:
This seems a bit strange, but if it's necessary to have all these mutations in one tick/render, then it's a way to go. I just don't think it's necessary that all these mutations happen in one render (they could already get batched, and the views should handle it in a performance-friendly way, rather than in middleware, but I don't feel too strongly.